src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -1733,7 +1734,7 @@ void QgsPostgresProvider::determinePrimaryKeyFromUriKeyColumn()
     if ( !mPrimaryKeyAttrs.isEmpty() )
     {
       bool unique = true;
-      if ( mCheckPrimaryKeyUnicity )
+      if ( mCheckPrimaryKeyUnicity || mOptions.trustDatasourceConfig )
```

Comment:
If I understand correctly, I would have used `|| !mOptions.trustDatasourceConfig` so that we do not check unicity if we trust the datasource ?

Reply:
right, fix